src/statemachine.rs

```diff
@@ -151,16 +151,27 @@ impl StateMachine {
             }
             Ok(DeviceCommand::Removed) => {
                 info!("Device {:?} was removed", dev.id());
+                send_status(
+                    status,
+                    crate::StatusUpdate::DeviceUnavailable {
+                        dev_info: dev.get_device_info(),
+                    },
+                );
                 return None;
             }
             Ok(DeviceCommand::Continue) => {
                 // Just continue
+                send_status(
+                    status,
+                    crate::StatusUpdate::DeviceSelected(dev.get_device_info()),
+                );
             }
             Err(_) => {
                 warn!("Error when trying to receive messages from DeviceSelector! Exiting.");
                 return None;
             }
         }
+        send_status(status, crate::StatusUpdate::SelectDeviceNotice);
```

Comment:
Hm, this causes Firefox's "Multiple devices available" prompt to be shown even when only one device is available. I'll attach another patch.